packages/components/organisms/f-checkout/src/mixins/validations.mixin.js

```diff
@@ -0,0 +1,27 @@
+export default {
+    methods: {
+        /**
+        * Returns true if `field` has been touched and if it is still empty
+        * The $dirty boolean changes to true when the user has focused/lost
+        * focus on the input field.
+        *
+        * @param component {string} The component being validated. (`Address` or `Guest`)
+        * @param field {string} The field to be checked.
+        */
+        isFieldEmpty (component, field) {
+            const type = `${component}Validations`;
+            return this.$v[type][field].$dirty && !this.$v[type][field].required;
+        },
+
+        /**
+        * Dispatches action `updateAddressDetails` or `updateCustomerDetails` to update input fields values in Vuex store.
+        *
+        * @param component {string} The component being updated. (`Address` or `Guest`)
+        * @param field {string} The field to be being updated. e.g. `firstName`, `line1`
+        * @param value {string} The new value.
+        */
+        updateFulfilmentDetails (component, field, value) {
```

Comment:
Not sure if I've missed something but was there an issue with just dispatching the action directly?

Reply:
I couldn't call them directly on @input because it needed the field and value to be sent.
It worked fine as a method in each component but it was very similar code being written twice

Reply:
Ignore me, really didn't need it in the end. It's removed now 😆